examples/credit_loan_decisions.py

```diff
@@ -0,0 +1,1359 @@
+metricframe_postprocess = MetricFrame(
+    metrics=fairness_metrics,
+    y_true=y_test,
+    y_pred=postprocess_pred,
+    sensitive_features=A_test,
+)
+
+# %%
+metricframe_postprocess.overall[metrics_to_report]
+
+# %%
+metricframe_postprocess.difference()[metrics_to_report]
+
+# %%
+# Now, let's compare the performance of our *thresholded* classifier with the
+# original *unmitigated* model.
+
+# %%
+compare_metricframe_results(
+    metricframe_unmitigated,
+    metricframe_postprocess,
+    metrics=metrics_to_report,
+    names=["Unmitigated", "PostProcess"],
+)
+
+# %%
+metricframe_postprocess.by_group[metrics_to_report].plot.bar(
+    subplots=True, layout=[3, 1], figsize=[12, 18]
+)
+
+
+# %%
+# We see that the `ThresholdOptimizer` algorithm achieves a much lower
+# disparity between the two groups compared to the _unmitigated_ model. However,
+# this does come with the trade-off that the `ThresholdOptimizer` achieves a
+# lower `balanced_accuracy` score for _male_ applicants.
+
+# %%
+# Reductions approach to unfairness mitigation
+# ============================================
+
+# %%
+# In the previous section, we took a fairness-unaware model and used the
+# `ThresholdOptimizer` to transform the model's decision boundary to satisfy our
+# fairness constraints. One key limitation of the `ThresholdOptimizer` is
+# needing access to our *sensitive_feature* during prediction time.
+
+# %%
+# In this section, we will use the *reductions* approach of `Agarwal et. al
+# (2018) <https://arxiv.org/pdf/1803.02453.pdf>`_ to produce models that satisfy
+# the fairness constraint without needing access to the sensitive features at
+# deployment time.
+#
+# The main reduction algorithm in Fairlearn is `ExponentiatedGradient`. The
+# algorithm creates a sequence of re-weighted datasets and retrains the wrapped
+# classifier on each of the datasets. This re-training process is guaranteed to
+# find a model that satisfies the fairness constraints while optimizing the
+# performance metric.
+#
+# The model returned by `ExponentiatedGradient` consists of several inner
+# models, returned by a wrapped estimator.
+
+# %%
+# To instantiate an `ExponentiatedGradient` model, we pass in two parameters:
+# - a base `estimator` (object that supports training)
+# - fairness `constraints` (object of type `fairlearn.reductions.Moment`)
+
+# %%
+# When passing in a fairness *constraint* as a `Moment`, we can specify an
+# `epsilon` value representing the maximum allowed difference or ratio between
+# our largest and smallest value. For example, in the below code,
+# `EqualizedOdds(difference_bound=epsilon)` means that we are using
+# `EqualizedOdds` as our fairness constraint, and we will allow a maximal
+# difference of `epsilon` between our largest and smallest *equalized odds*
+# value.
+
+# %%
+def get_expgrad_models_per_epsilon(
+    estimator, epsilon, X_train, y_train, A_train
+):
+    """
+    Instantiate and trains an ExponentiatedGradient model on the
```

Comment:
```suggestion
    Instantiates and trains an ExponentiatedGradient model on the
```

Reply:
In fact, perhaps :code:`ExponentiatedGradient` or even :class:`ExponentiatedGradient` are probably better.